p8s.go

```diff
@@ -41,8 +41,8 @@ func addRequest(labels map[string]string, bytes int) {
 }
 
 // InitMetrics sets up the prometheus registry and creates the metrics. Calling this
-// will reset any collected metrics
-func InitMetrics(additionalLabels ...string) {
+// will reset any collected metrics. Returns the registry so additional metrics can be registered.
+func InitMetrics(additionalLabels ...string) *prometheus.Registry {
```

Comment:
I don't like that. Instead, I'd prefer if the client provides the registry. Most clients would already have a registry defined. The library should be using that instead of defining a new one.

Reply:
https://martinfowler.com/bliki/TellDontAsk.html :)

Reply:
Not sure about that. This module is responsible for running the p8s we server too, so it's not like it's able to interact as simply with a passed in repo. If we wanted to allow for that then the module shouldn't really be running the `startPrometheusServer` function either, which would complicate the integration with our other code.

This can be done but it would be a major refactor.

Reply:
This module's API contract has grown organically as the scenarios it has needed to support have accumulated during the deployment to more Proxy Modules.

I'd be happy to do a second pass in the near future to revisit the API contract before we promote it a "publicly supported" (not the same as being public on GitHub and having capitalised identifiers in code).